Source/Scene/TimeDynamicPointCloud.js

```diff
@@ -0,0 +1,740 @@
+        var multiplier = clock.multiplier;
+        return isAnimating ? multiplier : 0.0;
+    }
+
+    function getIntervalIndex(that, interval) {
+        return that._intervals.indexOf(interval.start);
+    }
+
+    function getNextInterval(that, currentInterval) {
+        var intervals = that._intervals;
+        var clock = that._clock;
+        var multiplier = getClockMultiplier(that);
+
+        if (multiplier === 0.0) {
+            return undefined;
+        }
+
+        var averageLoadTime = that._getAverageLoadTime();
+        var time = JulianDate.addSeconds(clock.currentTime, averageLoadTime * multiplier, scratchDate);
+        var index = intervals.indexOf(time);
+
+        var currentIndex = getIntervalIndex(that, currentInterval);
+        if (index === currentIndex) {
+            if (multiplier >= 0) {
+                ++index;
+            } else {
+                --index;
+            }
+        }
+
+        // Returns undefined if not in range
+        return intervals.get(index);
+    }
+
+    function getCurrentInterval(that) {
+        var intervals = that._intervals;
+        var clock = that._clock;
+        var time = clock.currentTime;
+        var index = intervals.indexOf(time);
+
+        // Returns undefined if not in range
+        return intervals.get(index);
+    }
+
+    function reachedInterval(that, currentInterval, nextInterval) {
+        var multiplier = getClockMultiplier(that);
+        var currentIndex = getIntervalIndex(that, currentInterval);
+        var nextIndex = getIntervalIndex(that, nextInterval);
+
+        if (multiplier >= 0) {
+            return currentIndex >= nextIndex;
+        }
+        return currentIndex <= nextIndex;
+    }
+
+    function handleFrameFailure(that, uri) {
+        return function(error) {
+            var message = defined(error.message) ? error.message : error.toString();
+            if (that.frameFailed.numberOfListeners > 0) {
+                that.frameFailed.raiseEvent({
+                    url : uri,
+                    message : message
+                });
+            } else {
+                console.log('A frame failed to load: ' + uri);
+                console.log('Error: ' + message);
+            }
+        };
+    }
+
+    function requestFrame(that, interval, frameState) {
+        var index = getIntervalIndex(that, interval);
+        var frames = that._frames;
+        var frame = frames[index];
+        if (!defined(frame)) {
+            var transformArray = interval.data.transform;
+            var transform = defined(transformArray) ? Matrix4.fromArray(transformArray) : undefined;
+            var uri = interval.data.uri;
+            frame = {
+                pointCloud : undefined,
+                transform : transform,
+                timestamp : getTimestamp(),
+                sequential : true,
+                ready : false,
+                touchedFrameNumber : frameState.frameNumber
```

Comment:
I'm using terminology from the 3D Tiles code here. Basically it means a point cloud was "used" this frame. It could be the frame that was rendered or is loading.

I'm actually not sure where the name comes from historically or if its just made up, but it seems like an ok name.